scripts/announcements.R

```diff
@@ -0,0 +1,192 @@
+        return(invisible())
+    } else {
+        message("Event not posted to Issue.")
+    }
+}
+
+# Format as eg August 23
+day_month <- function(.date, add_name = TRUE) {
+    date_format <- "%B %e" # as August 23
+    if (add_name) date_format <- "%A, %B %e" # as Monday, August 23
+    trimws(format(as.Date(.date), format = date_format))
+}
+
+gh_issue_info <- function(.data) {
+    content <- .data %>%
+        mutate(needs_packages = ifelse(
+            !is.na(packages),
+            str_c(
+                "Please also install these packages: ",
+                str_replace_all(packages, " ", ", "), "."
+            ),
+            ""
+        )) %>%
+        mutate_at(vars(start_time, end_time), funs(strftime(., format = "%H:%M"))) %>%
+        glue_data(
+            "
+            {description}
+
+            - **Where**: MADLab at [Gerstein Science Information Centre](https://goo.gl/maps/2916Y54jQkx)
+            - **When**: {day_month(date)}, from {start_time}-{end_time} pm
+            - **Instructor**: TBA
+            - **Skill level**: {skill_level}
+
+            *Installation instructions*: You will need to install the appropriate programs. See the {program_language} section of the [installation instructions page](?link?). {needs_packages}
+
+            **Directions:**  MADLab is located in [Gerstein Science Information Centre](https://goo.gl/maps/2916Y54jQkx), Room B112 at the south end of the first lower level. Once you go through the main entrance of Gerstein, take a right turn down a corridor (across from the admin desk or just past the reading room), then take the stairs down and follow the signs to MADLab, the door should be open 10-15 minutes before the lesson.
+            "
+        )
+
+    .data %>%
+        mutate(content = content, title = str_c(title, " - ", day_month(date, add_name = FALSE))) %>%
+        select(title, content, skill_level, gh_labels)
+}
+
+create_gh_issues <- function(.data) {
+    .data %>%
+        gh_issue_info() %>%
+        pmap( ~ post_gh_issue(..1, ..2, c(..3, ..4)))
+}
+
+create_gh_issues(new_sessions)
+
+# Create files in _posts/ -------------------------------------------------
+# Adds the new sessions/events to the _posts folder.
+
+create_new_posts_with_content <- function(.data) {
+    new_post_filenames <-
+        glue_data(new_sessions, "{here::here('_posts')}/{date}-{key}.md")
+
+    # Get the GitHub Issue URL for the event.
+    gh_issue_number <- gh::gh("GET /repos/:owner/:repo/issues",
+                              owner = "uoftcoders",
+                              repo = "Events") %>%
+        map_dfr(~ data_frame(title = .$title, url = .$html_url)) %>%
+        mutate(title = str_remove(title, " - .* [1-9]?$"))
+
+    new_post_content <- .data %>%
+        left_join(gh_issue_number, by = "title") %>%
+        glue_data(
+            '
+            ---
+            title: "{title}"
+            text: "{description}"
+            location: "MADLab at Gerstein"
```

Comment:
Not sure if you want to hard code this location in. Let me know :)

Reply:
It can also be added to the events.csv database.

Reply:
we have monthly coffee and code in another place: grad house. but it's good to have it hard coded since it should link to the map

Reply:
Ok, I've changed it so the location is not hardcoded but you can still add a url/link to the map :)